pkg/controller/config/config_controller.go

```diff
@@ -280,6 +293,10 @@ func (r *ReconcileConfig) replayData(ctx context.Context, w *watch.Set) error {
 		for i := range u.Items {
 			syncKey := r.syncMetricsCache.GetSyncKey(u.Items[i].GetNamespace(), u.Items[i].GetName())
 
+			if r.configMatchSet.ExcludedNamespaces[match.Sync][u.Items[i].GetNamespace()] {
```

Comment:
This needs locks for thread safety, preferably embedded in a `configMatchSet` accessor method.

Reply:
added